lib/mozart_fetcher/component.ex

```diff
@@ -1,42 +1,80 @@
+    case Jason.decode(body, keys: :atoms) do
+      {:ok, data} ->
+        metric(config.id, 200)
+        %{index: component_index, id: config.id, status: 200, data: data}
+      {:error, _err} ->
+        metric(config.id, :json_decode_error)
+        failed_component(component_index, :json_decode_error, config.id, :ares)
+    end
+  end
+
   defp process(component_index, config, {:ok, %HTTPoison.Response{status_code: 200, body: body}}) do
-    ExMetrics.increment("success.component.process")
-    ExMetrics.increment("success.component.process.#{config.id}.200")
-    %Component{index: component_index, id: config.id, status: 200, envelope: Envelope.build(body)}
+    metric(config.id, 200)
+    %{index: component_index, id: config.id, status: 200, envelope: Envelope.build(body)}
   end
 
-  defp process(component_index, config, {:ok, %HTTPoison.Response{status_code: status_code, body: body}}) do
-    ExMetrics.increment("error.component.process")
-    ExMetrics.increment("error.component.process.#{config.id}.#{status_code}")
-    Stump.log(:error, %{message: "Non-200 response. Got status:#{status_code} for component #{config.id}"})
-    %Component{index: component_index, id: config.id, status: status_code, envelope: %Envelope{}}
+  defp process(component_index, config = %Config{format: "ares"}, {:ok, %HTTPoison.Response{status_code: status_code}}) do
+    metric(config.id, status_code)
+    %{index: component_index, id: config.id, status: status_code, data: %{}}
+  end
+
+  defp process(component_index, config, {:ok, %HTTPoison.Response{status_code: status_code}}) do
+    metric(config.id, status_code)
+    %{index: component_index, id: config.id, status: status_code, envelope: %Envelope{}}
+  end
+
+  defp process(component_index, config = %Config{format: "ares"}, {:error, %HTTPoison.Error{reason: reason}}) do
+    metric(config.id, reason)
+    failed_component(component_index, reason, config.id, :ares)
   end
 
   defp process(component_index, config, {:error, %HTTPoison.Error{reason: reason}}) do
-    ExMetrics.increment("error.component.process")
-    ExMetrics.increment("error.component.process.#{config.id}.#{reason}")
-    Stump.log(:error, %{message: "Failed to process HTTP request, reason: #{reason}"})
-    failed_component(component_index, reason, config.id)
+    metric(config.id, reason)
+    failed_component(component_index, reason, config.id, :envelope)
   end
 
   defp get(config) do
     LocalCache.get_or_store(config.endpoint, fn -> HTTPClient.get(config.endpoint) end)
   end
 
-  defp failed_component(component_index, :timeout, id) do
-    %Component{index: component_index, id: id, status: 408, envelope: %Envelope{}}
+  defp failed_component(component_index, :timeout, id, :envelope) do
+    %{index: component_index, id: id, status: 408, envelope: %Envelope{}}
+  end
+
+  defp failed_component(component_index, _, id, :envelope) do
+    %{index: component_index, id: id, status: 500, envelope: %Envelope{}}
```

Comment:
I wonder if at a later date we should just return envelopes as the data key and we use the upstream service to check for envelope keys or data. If we trust fetcher more, we won't necessarily need to do a lot of validation.

Reply:
I agree, this extra `data:` key only really exists because I found Requester hard to change